src/net_processing.cpp

```diff
@@ -1512,6 +1512,11 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
     std::deque<CInv>::iterator it = pfrom->vRecvGetData.begin();
     std::vector<CInv> vNotFound;
     const CNetMsgMaker msgMaker(pfrom->GetSendVersion());
+
+    // Note that if we receive a getdata for a MSG_TX or MSG_WITNESS_TX from a
+    // block-relay-only outbound peer, we will stop processing further getdata
+    // messages from this peer (likely resulting in our peer eventually
```

Comment:
Note that this comment is incomplete, we'll stop processing *all* messages from this peer (including pongs, so we'll eventually disconnect them as well).

Reply:
I agree that we should update this comment (see https://github.com/bitcoin/bitcoin/pull/15759#discussion_r314786257)